src/functions/RepresentableFunction.h

```diff
@@ -74,4 +76,15 @@ template <int D> class RepresentableFunction {
     virtual bool isZeroOnInterval(const double *a, const double *b) const { return false; }
 };
 
+/*
+ * Same as RepresentableFunction, but output a matrix of values
+ * for all points in a node, given its NodeIndex.
+ *
+ */
+class RepresentableFunction_M {
+public:
+    RepresentableFunction_M() {}
+    virtual Eigen::MatrixXd evalf(mrcpp::NodeIndex<3> nIdx) const = 0;
+};
```

Comment:
Is this class actually used? I see there's a `FunctionTree::integrateEndNodes` function taking such an argument, but I can't find an actual implementation of any `RepresentableFunction_M`

Reply:
Not yet: My original plan was to make a representableFunction_M for representing the XCpotential, so that the entire potential never needed to be stored. However I did not use it at the end. I still think it may/will be useful: in general it is much more efficient to treat all points of a node together, rather than one by one.